openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiEnumSpec.groovy

```diff
@@ -1,12 +1,134 @@
 package io.micronaut.openapi.visitor
 
 import io.micronaut.openapi.AbstractOpenApiTypeElementSpec
+import io.micronaut.openapi.swagger.PrimitiveType
 import io.swagger.v3.oas.models.OpenAPI
 import io.swagger.v3.oas.models.Operation
 import io.swagger.v3.oas.models.media.Schema
 
 class OpenApiEnumSpec extends AbstractOpenApiTypeElementSpec {
 
+    void "test build OpenAPI custom enum jackson JsonValue method schema type"() {
+
+        when:
+        buildBeanDefinition('test.MyBean', '''
+package test;
+
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+@Controller
+class OpenApiController {
+    @Get("/currency/{currency}")
+    public void postRaw(MyEnum myEnum) {
+    }
+}
+
+enum MyEnum {
+
+    @JsonProperty("1")
+    FIRST(1),
+    @JsonProperty("2")
+    SECOND(2),
+    ;
+
+    private final int value;
+
+    MyEnum(int value) {
+        this.value = value;
+    }
+
+    @JsonValue
```

Comment:
This test is confusing to me. What are we testing @JsonProperty or @JsonValue?

should not we have two tests?

```
enum MyEnum {
     FIRST(1),
     SECOND(2) ;

     private final int value;

     MyEnum(int value) {
         this.value = value;
     }

     @JsonValue
     public int getValue() {
         return value;
     }
}
```` 

```
enum MyEnum {
    @JsonProperty("1")
     FIRST(1),
    @JsonProperty("2")
     SECOND(2) ;
}
````

Reply:
@sdelamo No, you missunderstood. We need `@JsonProperty` annotation to set enum values which different with enum value name. This is single way to customize enum values. In tests we need to set int values for every enum value and annotation processor haven't  acccess to enum valus objects, only to structure and that's why we need to add annotation JsonProperty

Reply:
> We need @JsonProperty annotation to set enum values which different with enum value name.

But you are using the exact value not a different one. 

The test has: 

```
    @JsonProperty("1")
     FIRST(1),
     @JsonProperty("2")
     SECOND(2),
```

It does not have:


```
    @JsonProperty("5")
     FIRST(1),
     @JsonProperty("6")
     SECOND(2),
```


Moreover, are my examples not how people will write code in their applications? You will never write an enum with both `@JsonProperty` and `@JsonValue`  right?

Reply:
@sdelamo not exact. Value for enum value `FIRST` will be "FIRST", not 1. Again: annotation process can't see parameters in construct that you set. In other words, micronaut openapi don't see constructor argument value 1 or 2, micronaut-openapi can see only annotaions parameters

Reply:
@sdelamo 
```java
@JsonProperty("1")
FIRST(1),
@JsonProperty("2")
SECOND(2)
```
This need to set number values for FIRST and SECOND enum values. Because this is sungle way to set numbers to enum values in annotation processor. The other way is much more complicated and nned to run the compiler directly from the annotation processor

Reply:
And yes, this is a limitation. You will have to declare both JsonProperty and JsonValue if you want to use micronaut-openapi . These are the limitations of the annotation processor. Another way is to explicitly specify the type in the @Schema annotation.

For example I'm using JsonValue and JsonProperty in the generated code by hud schema via xjc .